### PR TITLE
fix: remote transcription posts to /transcribe, not /align (stem-splitter#17)

### PR DESCRIPTION
**Pull request by topkoa**
Client half of [feedBack-plugin-stem-splitter#17](https://github.com/got-feedBack/feedBack-plugin-stem-splitter/issues/17). Pairs with [feedBack-demucs-server#14](https://github.com/got-feedBack/feedBack-demucs-server/pull/14), which adds the endpoint — **merge that one first.**

## Remote transcription has never worked

`transcribe_vocals_remote()` POSTs the vocal stem to **`/align`**. That endpoint is *forced alignment* — *"here are the lyrics, tell me when each word is sung"* — and its `text` field is **required**. We have no lyrics; transcribing them is the entire point.

So the server rejected every request with a **422 from FastAPI's validation layer, before its handler ever ran.** Not a regression — this path has never worked for anybody.

## Three fixes

1. **POST to `/transcribe`**, which takes only the audio.

2. **`language` moves from the query string to the form body**, where the server actually reads it (`Form("")`). As a query param it was *silently ignored* — so an explicit hint did nothing, Whisper's auto-detection quietly decided instead, and the wrong wav2vec2 aligner got loaded. It "worked", it was just wrong: the failure mode that hides for months.

3. **Error bodies are no longer cut at 300 chars.** The body *is* the diagnosis — a 422's JSON names the field it rejected, a 500's traceback answers on its **last** line. Both got decapitated, which is part of why this stayed invisible: the message explaining the bug was inside the part that got cut.

## Why nothing caught it

Every test of this module tested the **mapper** (`_whisperx_to_sloppak`), fed a hand-written dict. The mapper was always fine. **The request was never exercised, and the request was the bug.**

`tests/test_lyrics_transcribe_remote.py` now pins the request itself: the endpoint, the form field, the multipart upload, the bearer token, an instrumental returning no lyrics rather than an error, and a 404 carrying "the server is too old" through instead of swallowing it.

Verified the two key tests **fail** against the old `/align` + `params` call.

## Compatibility

A server predating `/transcribe` answers 404, and the error now says so plainly rather than being truncated into noise.

Lyrics suite green (25). The two `test_builtin_diagnostic_seed` symlink failures are pre-existing on `main` — a Windows symlink-privilege artefact, unrelated to this change.

## Summary by CodeRabbit

* **New Features**
  * Updated remote vocal transcription to use the current transcription endpoint.
  * Language hints are now sent as a form field (only when provided).
  * Added support for sending API-key authentication with remote requests.
  * Instrumental/no-singing responses now return empty lyric results.

* **Bug Fixes**
  * Improved error handling by including informative server error details and applying safe truncation for large error bodies, including clearer 404 messaging.

* **Tests**
  * Added regression tests covering endpoint targeting, request/form encoding, multipart upload, authentication header behavior, and error-body/truncation edge cases.

### CODE REVIEW

**Comment by topkoa** [comment]
@coderabbitai review

File: lib/lyrics_transcribe.py [hunk 1]
```diff
@@ -454,21 +481,34 @@ def transcribe_vocals_remote(
...
             files={"file": (vocals_path.name, f, "audio/ogg")},
-            params=params,
+            data=form or None,
             headers=headers or None,
             timeout=timeout,
         )
 
     if resp.status_code != 200:
-        raise RuntimeError(f"WhisperX server error ({resp.status_code}): {resp.text[:300]}")
+        raise RuntimeError(f"WhisperX server error ({resp.status_code}): {_err_body(resp)}")
```
[the remark this answers is not in the document]
Comment (topkoa) [reply lib/lyrics_transcribe.py:554]: Fixed — and it's the worst place to leave it stale: the docstring is where a reader goes to find out what this function talks to, and it was still naming the endpoint the PR exists to stop talking to.

It now says it posts to `/transcribe`, why `/align` was wrong (its `text` field is required; we have no lyrics, producing them is the point), and that an older server answers 404.

File: lib/lyrics_transcribe.py [hunk 2]
```diff
@@ -416,6 +425,24 @@ def transcribe_vocals_local(
...
+    This was capped at 300 chars, which is enough for "Internal Server Error" and not much else.
+    The bodies carrying the most diagnosis are the long ones — a FastAPI validation body naming
+    the field it rejected, a 500 whose traceback answers on its LAST line — and those are exactly
+    the ones a 300-char cap decapitates. The cap survives so a server answering with a 2 MB HTML
+    error page can't dump a novel into a log line.
+    """
+    text = getattr(resp, "text", "") or ""
+    if len(text) <= _MAX_ERR_BODY:
+        return text.strip()
+    return text[:_MAX_ERR_BODY].strip() + f"\n… [truncated, {len(text)} chars total]"
```
[the remark this answers is not in the document]
Comment (topkoa) [reply lib/lyrics_transcribe.py]: Both real, and the first one is a bound that isn't one:

```
>>> len(_err_body(resp_with_500k_body))
4014      # for a _MAX_ERR_BODY of 4000
```

The marker was appended *after* the slice. A cap that's advisory surprises precisely the callers who trust it — a log line, or a job record that gets persisted to disk and re-read on every load. The marker now fits inside the bound.

The strip-ordering one is subtler and I'm glad you flagged it: measuring before stripping means a 300-char JSON body followed by kilobytes of trailing whitespace reads as "too long" and gets cut — **real content sacrificed to make room for blanks**, in a function whose entire job is not losing the diagnosis. Strip first, then measure. Both pinned by tests.

**Comment by topkoa** [comment]
@coderabbitai review

File: lib/lyrics_transcribe.py [hunk 3]
```diff
@@ -416,6 +425,31 @@ def transcribe_vocals_local(
...
+    # is not a long body, and truncating it would cut real content to make room for blanks.
+    text = (getattr(resp, "text", "") or "").strip()
+    if len(text) <= _MAX_ERR_BODY:
+        return text
+    # The marker has to fit INSIDE the cap, not be appended past it — otherwise _MAX_ERR_BODY is
+    # a suggestion rather than a bound, and the one caller who trusts it (a log line, a job record
+    # persisted to disk) is the one who gets surprised.
+    marker = f"\n… [truncated, {len(text)} chars total]"
+    keep = max(0, _MAX_ERR_BODY - len(marker))
+    return text[:keep].rstrip() + marker
```
[the remark this answers is not in the document]
Comment (topkoa) [reply lib/lyrics_transcribe.py]: This is the best catch of the round: **the docstring said the diagnosis is on the last line, and the code kept the first 4000 chars.** It cut off exactly the part it exists to preserve — the same mistake as the 300-char cap it was replacing, one level up. A window holding `Traceback (most recent call last)` and none of the exception is a window onto nothing.

Head **and** tail now, both inside the bound: two thirds head (what was being attempted), one third tail (what actually went wrong), marker between them. Verified the test fails against head-only truncation.

**Comment by topkoa** [comment]
@coderabbitai review

File: lib/lyrics_transcribe.py [hunk 4]
```diff
@@ -454,21 +505,34 @@ def transcribe_vocals_remote(
...
-        params["language"] = language
+        form["language"] = language
 
     with open(vocals_path, "rb") as f:
         resp = requests.post(
-            f"{server_url}/align",
+            f"{server_url}/transcribe",
             files={"file": (vocals_path.name, f, "audio/ogg")},
-            params=params,
+            data=form or None,
```
[the remark this answers is not in the document]
Comment (topkoa) [reply lib/lyrics_transcribe.py]: Both good, and the first has teeth: the caller (`_maybe_transcribe_lyrics`) catches `RuntimeError` specifically, so one song's failed lyrics don't take down the batch around it. A DNS failure or a timeout coming out as `requests.RequestException` **walks straight past that handler** — turning "this song's lyrics failed" into "the whole batch died", which is a much worse day for someone converting an album.

Every exit is a `RuntimeError` now: transport errors, an unreadable stem, and non-200s. Tests cover connection failure, timeout, and a missing stem file.

The 404 explains itself too — a bare "404" sends someone hunting for a typo in their server URL, when the real answer is that their server predates `/transcribe` and only we are in a position to know that.

**Comment by topkoa** [comment]
@coderabbitai review
